cinema/tests/test_movie.py

```diff
@@ -0,0 +1,192 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieListSerializer, MovieDetailSerializer
+
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=(movie_id,))
+
+
+def sample_movie(**params) -> Movie:
+    default_movie = {
+        "title": "Interstellar",
+        "description": "A movie about future in space.",
+        "duration": 169,
+    }
+    default_movie.update(params)
+    return Movie.objects.create(**default_movie)
+
+
+class UnauthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test_user@example.com", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movies_list(self):
+        sample_movie()
+
+        movie_with_genres = sample_movie()
+        genre_1 = Genre.objects.create(name="fiction")
+        genre_2 = Genre.objects.create(name="drama")
+        movie_with_genres.genres.add(genre_1, genre_2)
+
+        res = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_movies_by_genres(self):
+        movie_without_genres = sample_movie()
+        movie_with_genre_1 = sample_movie(title="Movie 1")
+        movie_with_genre_2 = sample_movie(title="Movie 2")
+
+        genre_1 = Genre.objects.create(name="fiction")
+        genre_2 = Genre.objects.create(name="drama")
+
+        movie_with_genre_1.genres.add(genre_1)
+        movie_with_genre_2.genres.add(genre_2)
+
+        res = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"}
+        )
+
+        serializer_without_genres = MovieListSerializer(movie_without_genres)
+        serializer_movie_genre_1 = MovieListSerializer(movie_with_genre_1)
+        serializer_movie_genre_2 = MovieListSerializer(movie_with_genre_2)
+
+        self.assertIn(serializer_movie_genre_1.data, res.data)
+        self.assertIn(serializer_movie_genre_2.data, res.data)
+        self.assertNotIn(serializer_without_genres, res.data)
```

Comment:
The `assertNotIn` method should compare `serializer_without_genres.data` with `res.data`, not `serializer_without_genres`.